otri/downloader/__init__.py

```diff
@@ -808,3 +817,413 @@ def _post_process(self, atom: Mapping, **kwargs) -> Mapping:
+        '''
+        Appends a component at the end of the components list.
+
+        Parameters:
+            component : AdapterComponent
+                Component to append.
+        '''
+        self.components.append(component)
+
+    def download(self, o_stream: WritableStream, **kwargs) -> LocalStream:
+        '''
+        Retrieves some data from a source.
+        Each component is called in the given order.
+
+        Parameters:
+            o_stream : WritableStream
+                Stream where to output parsed data. Should NOT be already closed.
+
+            Other parameters depend on what components the adapter uses.
+        Returns:
+            A closed stream of atoms, the same object as parameters o_stream.
+        '''
+        if o_stream.is_closed() and not self.allow_o_stream_closed:
+            raise AttributeError("o_stream cannot be already closed")
+
+        data_stream = LocalStream()  # Connecting pipe between data retrieval and atomization
+        for comp in self.components:
+            kwargs.update(comp.prepare(**kwargs) or {})
+
+        # TODO: delet dis
+        log.d(f"Preparation: {kwargs}")
+
+        for comp in self.components:
+            kwargs.update(comp.retrieve(data_stream=data_stream, **kwargs) or {})
+
+        if self.sync:
+            # Close data stream after all downloads are performed, no more data can be added
+            data_stream.close()
+
+        log.d(f"Retrieval: {kwargs}")
+
+        for comp in self.components:
+            kwargs.update(comp.atomize(data_stream=data_stream, output_stream=o_stream, **kwargs) or {})
+
+        if self.sync:
+            o_stream.close()
+
+        log.d(f"Atomization: {kwargs}")
+
+        return o_stream
+
+
+class TickerSplitterComp(AdapterComponent):
+    '''
+    Uses preparation phase to split a ticker list is multiple lists of a maximum size.
+    Only uses prepare method.
+    '''
+
+    def __init__(self, max_count: int = 1, tickers_name: str = "tickers", out_name: str = "ticker_groups"):
+        '''
+        Parameters:
+            max_count : int
+                Positive number for maximum number of tickers allowed per group/list/data request.
+            tickers_name : str
+                Name for ticker list parameter.
+            out_name : str
+                Name for the output ticker groups.
+        '''
+        self._max_count = max_count
+        self._tickers_name = tickers_name
+        self._out_name = out_name
+
+    def prepare(self, **kwargs):
+        # Checks
+        if self._tickers_name not in kwargs:
+            raise ValueError(f"Missing '{self._tickers_name}' parameter")
+        if not isinstance(kwargs[self._tickers_name], Iterable):
+            raise ValueError("'{self._tickers_name}' parameter is not iterable, it's {type(kwargs[self._tickers_name])}")
+
+        kwargs[self._out_name] = [kwargs[self._tickers_name][i:i + self._max_count]
```

Comment:
I came back to this twice before understanding that it is splitting a list (of streams??). There has to be something in `itertools` or somewhere else to do this in a clearer way.

Reply:
What it's doing is splitting a list in chunks of size n (except for the last one) so just like
`x = [l[i:i + n] for i in range(0, len(l), n)] `
The names are indeed pretty long and it's hard to read so I should rename something.

Reply:
I remembered correctly saying there are methods to slice a list in (almost) equal parts, but couldn't find one for slicing into fixed length parts. By the way, are you sure this raises no error? If I have a max count of 5, and 13 tickers, wouldn't this crash when taking the third slice?

Reply:
Nope it won't crash, when slicing if the second operator is higher than the length of the list it'll automatically take it up to the list length

```py
l=[1,2,3,4,5]
print(l[3:200]) # prints [4,5]
```

Reply:
til